### PR TITLE
chore(metrics): Add isAuthenticated flag

### PR DESCRIPTION
**Pull request by underyx**
PR checklist:

- [x] Documentation is up-to-date
- [x] Changelog is up-to-date
- [x] Change has no security implications (otherwise, ping security team)

### CODE REVIEW

File: PRIVACY.md [hunk 1]
```diff
@@ -134,12 +134,13 @@ r2c will:
...
 |             | Config hash               | One-way hash of the config argument                                    | Understand performance and accuracy improvements                                           | `ede96c41b57de3e857090fb3c486e69ad8efae3267bac4ac5fbc19dde7161094`                                                                                                                    | String         |
+|             | Is authenticated          | Whether the user is logged in                                          | Understand popularity of logged in features                                                | `false`                                                                                                                                                                               | Boolean        |
```
Comment (spencerdrak) [inline PRIVACY.md]: This may require some sort of notification to the community, even though it's kinda implied that this is happening already via tokens.

Comment (underyx) [inline PRIVACY.md]:
```suggestion
|             | Is authenticated          | Whether the user logged in to semgrep.dev with `semgrep login`          | Understand popularity of logged in features                                                | `false`                                                                                                                                                                               | Boolean        |
```

**Review by spencerdrak — COMMENTED** [review]
Aside from the timezone issue discuss with @underyx, LGTM.

File: PRIVACY.md (hunk 1 above)
Comment (underyx) [reply PRIVACY.md]: done https://r2c-community.slack.com/archives/CK86BJ5DW/p1652816048837079

**Review by spencerdrak — APPROVED** [review]

**Review by holdenmcg — APPROVED** [review]

**Pull request merged**